bigchaindb/models.py

```diff
@@ -352,11 +361,15 @@ def from_db(cls, bigchain, block_dict, from_dict_kwargs=None):
         """
         asset_ids = cls.get_asset_ids(block_dict)
         assets = bigchain.get_assets(asset_ids)
+        txn_ids = cls.get_txn_ids(block_dict)
+        metadata = bigchain.get_metadata(txn_ids)
+        # reconstruct block
         block_dict = cls.couple_assets(block_dict, assets)
+        block_dict = cls.couple_metadata(block_dict, metadata)
         kwargs = from_dict_kwargs or {}
         return cls.from_dict(block_dict, **kwargs)
 
-    def decouple_assets(self):
+    def decouple_assets(self, block_dict=None):
```

Comment:
Why do we need that additional optional parameter?

Reply:
this is done in order to make the interface same as `decouple_metadata`.